[init] Icon 기초 세팅

kamja0510 opened the pull request:
## Related issue 🛠
- closed #3 

## Work Description ✏️
- 디자이너 선생님들이 정리해주신 아이콘 삽입
- gui와 비교하여 정리되지 않은 아이콘 확인 : [피그마링크](https://www.figma.com/design/iYoE09THjUlkPynNQEYVIT/%F0%9F%8D%9EANDROID-%EC%9E%91%EC%97%85-%EA%B3%B5%EA%B0%84%F0%9F%8D%9E?node-id=250-63583&p=f&t=Oe9yReRHw1R3adDp-0) -> additional section

## Screenshot 📸
- 영상 필수첨부

https://github.com/user-attachments/assets/4f21cde6-0f45-40d1-afa3-415b6ddb6dd1



## Uncompleted Tasks 😅

## To Reviewers 📢
네이밍 컨벤션이 `ic_사용처_설명_크기` 이렇게 되어있는데 사용되는 곳은 공통인 것이 너무 많아 뺐고, 나중에 확장성 까지 고려해서 네이밍을 `ic_설명_속성_크기` 이렇게 해보았습니다.

속성은 아무것도 없는 것을 default로 표기해놓았습니다.

속성 중 tight 속성이 있는데 이 속성은 가로가 세로의 반 크기로 지정되는 것 같습니다. 따라서 ic_home_tight_24라는 아이콘이 있으면 가로는 12px일 것입니다.

피그마 링크에 정리되지 않은 아이콘은 어떻게 하면 좋을까요~~

##수정사항 
네이밍 : `ic_기본모양_세부모양(optional)_사이즈특징(두께, 크기 등, 필수(기본값 : default))_사이즈(정수, 필수)`

kamja0510 committed:
[init] #3 디자이너분들이 정리해놓은 icon 세팅

kamja0510 committed:
[merge] #3 theme 세팅 병합

HAJIEUN02 commented on app/src/main/res/drawable/ic_check_default_24.xml:
`p4`
`ic_check_thick_24` 이 친구와 같은 아이콘으로 기대되는데 전혀 다른 모양의 아이콘이기 때문에 네이밍을 수정하는 게 좋을 것 같아요!
<img width="320" alt="image" src="https://github.com/user-attachments/assets/a09e3dac-1633-477c-a0bf-d7743632506a" />

요 `ic_check_default_24` 아이콘이 디자이너에서는 "페이지 조각" 이라고 부르거든요? 그래서 `ic_piece_default_24`라든지 `ic_page_default_24`라든지.. 그런 식으로 수정하면 좋을 것 같아요~

HAJIEUN02 commented on app/src/main/res/drawable/ic_apple_default_24.xml:
`p4` 안드로이드에서는 애플 로그인 구현할 필요가 없어서~ 지워주심 될 것 같아요

kamja0510 replied:
넵

HAJIEUN02 commented:
`p2`
<img width="473" alt="image" src="https://github.com/user-attachments/assets/7883fc60-a7a0-4c92-98e3-9939ce9c6b49" />

이 아이콘이 안 들어가 있는 듯합니다~

HAJIEUN02 commented:
<img width="127" alt="image" src="https://github.com/user-attachments/assets/fdfea5ed-cc8c-478c-9025-e352429310e0" />
얘두 안 보임다

HAJIEUN02 commented on app/src/main/res/drawable/ic_home_variant2_24.xml:
`p4`
피그마 네이밍을 전부 그대로 따라갈 필요는 없습니다!
variant2는 다른 네이밍들(thick, small 등의 구체적인 속성)과 컨벤션이 맞지 않으니 속성 위주로 네이밍해주세요.
그리고 얘는 예전 바텀네비 아이콘이라 삭제해도 돼용

kamja0510 replied:
예를 들면 똑같이 집 모양의 아이콘이라도 변형된 구조의 아이콘들이 있을텐데 그런 것들은 thick,small 같은 속성으로 나타내기 힘드니 variant라는 속성이 와닿긴했습니다.

HAJIEUN02 commented on app/src/main/res/drawable/ic_x_o_24.xml:
`p4`
`ic_x_circle_24`로 수정하는게 좋겠네용.
다른 파일에서는 `ic_plus_circle_24`, `ic_plus_thick_24` 이런 식으로 네이밍 하셨으니까요!

HAJIEUN02 commented on app/src/main/res/drawable/ic_bellringing_default_24.xml:
`p5` bell_ringing으로 underscore 하나 넣어주시면 좋을 것 같습니다

HAJIEUN02 commented on app/src/main/res/drawable/ic_user_1_24.xml:
`p4` 숫자를 넣으면 다른 속성과 오해가 생길 수 있어서 `ic_user_one_24` 이런 식으로 수정해주심 좋을 듯해여

HAJIEUN02 commented on app/src/main/res/drawable/ic_user_2_24.xml:
`p4` 숫자를 넣으면 다른 속성과 오해가 생길 수 있어서 `ic_user_two_24` 이런 식으로 수정해주심 좋을 듯해여

HAJIEUN02 commented on app/src/main/res/drawable/ic_infocircle_default_24.xml:
`p4` 다른애들은 plus_circle 이런 식으로 분리해줬는데 얘는 infocircle 이렇게 붙어있어가지구..
레전드 강박증 여성.. .. ... ... ㅠㅠ 얘도 ic_info_circle.. 이런 식으로 수정해주심 좋을 것 같아요!
아니면 싹다 붙이든지!! 하나로 통일....... plz

kamja0510 replied:
일일이 다 설명하기가 힘든데 plus_circle인 친구는 아이콘 이름이 plus인데 속성이 circle인 친구이고
infocircle은 아예 아이콘 이름이 infocircle인 것을 명시한 것입니다!

HAJIEUN02 commented on app/src/main/res/drawable/ic_circlecheck_default_24.xml:
`p5` 개발 시 checkcircle과 circlecheck를 매번 헷갈릴 것 같아서 네이밍을 다시 한번 고민해보고 수정해주시면 좋을 듯해요!! 이 부분은 사용자가 아니라 우리를 위해 네이밍 하는거니까.. 우리가 잘 알아볼 수 있을 네이밍을 생각해보아요

HAJIEUN02 commented on app/src/main/res/drawable/ic_messagebox_messagetextsquare02_24.xml:
`p4` 얘는 messagetextsquare02가 너무 줄줄이소세지라!!!!!! 피그마 따라갈 필요 없이 `ic_messagebox_text_24` 이거 어떠심까?

HAJIEUN02 commented on app/src/main/res/drawable/ic_kakao_variant2_24.xml:
`p3` 이거 삭제해주세요! `ic_userplus_default_24`와 같은 아이콘입니다

kamja0510 replied:
꼼꼼함 최고다..
kakao로 바꾸겠습니다

HAJIEUN02 commented on app/src/main/res/drawable/ic_file_variant2_24.xml:
`p4` variant2.보단 bnv_todo라든지 file_check 이런 식으루 하심 좋을 것 같네요

HAJIEUN02 commented:
`p2`
<img width="117" alt="image" src="https://github.com/user-attachments/assets/2b97464e-4b88-4cd7-b1a2-cf014aa20e50" />
얘두 안 보임핑!

HAJIEUN02 reviewed (commented):
모든 속성이 False인 경우를 default로 하셨군요.
개발할 때 보기 쉽게 피그마 기반 네이밍 꼼꼼하게 잘 해주신 것 같아서 죠습니다!
`ic _ 어쩌구 _ 크기 컨벤션` 을 기본으로 유도리 있게 컨벤션은 조정해도 되어요.
우리 팀 네이밍 컨벤션 페이지에 수정해둘게요~!

맞다 그리구 늘 확장성을 고려하는 것은 좋으나 불필요하게 너무 먼 미래까지 보는 건 아닐지 생각해보는 것도 좋을 것 같아요! 예를 들면 바텀네비에만 쓰이는 아이콘 4개는 thick small 등등 추가 속성이 생길 가능성이 굉장히 낮잖아요. 그래서 얘네들은 `ic_bnv_subject_24` 이런 식으로 네이밍한다면 가독성이 더 좋지 않을까? 라는 생각이 들어욥. 이 내용은 리뷰에 추가 안해두었지만 고민해보셔요

아이콘 많아서 귀찮았을텐데 고생하셨습니다~!

HAJIEUN02 commented:
`p2` 자물쇠 아이콘 추가부탁!
default plus 버튼도용
<img width="882" alt="image" src="https://github.com/user-attachments/assets/6f5d5aaf-c309-4f33-bdd4-c3315df5d8a1" />

beom84 commented:
피그마에서 체크해봤는데 지은이가 말한거 이외에는 . 잘추가되어있는것 같습니다! 고생하셨습니다!!!

kamja0510 committed:
[init] #3 icon 세팅 코드리뷰 반영

쓰지않는 아이콘 제거, 정리는 안되어있지만 쓰이는 아이콘 추가, 전체 적인 아이콘 네이밍 수정
네이밍 : ic_기본모양_세부모양(optional)_사이즈특징(두께, 크기 등, 필수(기본값 : default))_사이즈(정수, 필수)

kamja0510 committed:
[init] #3 icon 세팅 코드리뷰 반영

쓰지않는 아이콘 제거, 정리는 안되어있지만 쓰이는 아이콘 추가, 전체 적인 아이콘 네이밍 수정
네이밍 : ic_기본모양_세부모양(optional)_사이즈특징(두께, 크기 등, 필수(기본값 : default))_사이즈(정수, 필수)

HAJIEUN02 commented on app/src/main/res/drawable/ic_checkcircle_default_24.xml:
check_circle이랑 checkcircle이랑 진짜 개헷갈린다ㅠ 그치만 이게 최선인듯. . 하암🥹😭😭😭

kamja0510 replied:
잠깐만 이건 실수

kamja0510 replied:
아 아니네요 ㅎㅎ 진짜 헷갈리네

HAJIEUN02 reviewed (approved):
고생했더용~! 
앞으로도 얘기 많이 하면서 우리 팀만의 규칙들을 많이 만들어 나가 봅시다
